### PR TITLE
nodefs: delay directory read till ReadDir

### PR DESCRIPTION
**Pull request by rfjakob**
We used to read the directory already on OpenDir, which means that
we missed changes in the directory between a user's open() and
readdir() call.

https://github.com/hanwen/go-fuse/issues/252

### CODE REVIEW

**Comment by navytux** [comment]
@rfjakob, could we please have a unit test for this fix?

**Comment by rfjakob** [comment]
Agreed. Will add.

File: fuse/nodefs/dir.go [hunk 1]
```diff
@@ -13,33 +13,30 @@ import (
 
 type connectorDir struct {
 	node  Node
+	inode *Inode
 	rawFS fuse.RawFileSystem
 
 	// Protect stream and lastOffset.  These are written in case
```
Comment (slackner) [inline fuse/nodefs/dir.go]: There is also a remaining reference to `lastOffset`. We might want to update that comment aswell.

**Comment by hanwen** [comment]
if we do this, OpenDir failures will manifest on readdir() rather than open(). Is that a problem?

**Comment by rfjakob** [comment]
Thing is, the kernel issues a lookup before opendir, so this will happen rareley.

**Comment by rfjakob** [comment]
Tests added! (travis failures are from unionfs)

**Comment by navytux** [comment]
@rfjakob, thanks for the test.

**Comment by rfjakob** [comment]
@hanwen done

**Comment by hanwen** [comment]
thanks, I rebased and pushed.

**Pull request closed without merging**